src/core/models.py

```diff
@@ -454,3 +469,43 @@ def extra_user_roles(self) -> str:
         When `admin` is set, the Kafka charm interprets this as a new super.user.
         """
         return self.relation_data.get("extra-user-roles", "")
+
+
+class Balancer(RelationState):
+    """State collection for balancer."""
+
+    def __init__(
+        self,
+        relation: Relation | None,
+        data_interface: DataPeerData,
+        substrate: Substrates,
+    ):
+        super().__init__(relation, data_interface, None, substrate)
+        self.data_interface = data_interface
+
+    @property
+    def rack_aware(self) -> bool:
+        """Is the Kafka deployment rack aware?"""
+        if not self.relation:
+            return False
+
+        return (
+            self.data_interface.fetch_my_relation_field(
+                relation_id=self.relation.id, field="rack-aware"
+            )
+            == "true"
+            or False
+        )
```

Comment:
Maybe I was doing something wrong with the data interface because I would get the `"false"` str which evaluates to _true-ish_.
Any ideas where things went wrong?

Reply:
I think you can just remove the `or False` L497, which should still evaluate as a boolean.

Reply:
Removed in  671643ebf594523a7a8d621551d69706d88f10d8, sorry I did not understand the first time